### PR TITLE
Weavegather torch

### PR DESCRIPTION
**Pull request by NimishaDey**
## Description

This PR is to port Weave Gather to PyTorch.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by rbharath — COMMENTED** [review]
@ARY2260 can you do a round of review on this PR? I will do my pass after

File: deepchem/models/tests/test_weave_gather.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    assert len(outputs) == 2
+    assert np.array(outputs[0]).shape == (11 * 75,)
+    assert np.array(outputs[1]).shape == (11 * 75,)
+    assert np.allclose(
+        outputs.numpy(),
+        np.load(
+            "deepchem/models/tests/assets/weavegather_results_without_compression.npy"
+        ),
+        atol=1e-4)
+    # Try with compression
```
Comment (ARY2260) [inline deepchem/models/tests/test_weave_gather.py:50]: write this test in a separate function

**Review by ARY2260 — CHANGES_REQUESTED** [review]
LGTM, once the test splitting is completed

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**